server/engine/src/main/java/org/eclipse/lsp/cobol/service/AnalysisService.java

```diff
@@ -45,19 +52,25 @@ public class AnalysisService {
   private List<String> copybookExtensions;
 
   private final DocumentModelService documentService;
+  private final CFASTBuilder cfastBuilder;
+  private final Provider<CobolLanguageClient> clientProvider;
 
   @Inject
   AnalysisService(
       LanguageEngineFacade engine,
       ConfigurationService configurationService,
       @Named("combinedStrategy") CopybookIdentificationService copybookIdentificationService,
       CopybookService copybookService,
-      DocumentModelService documentService) {
+      DocumentModelService documentService,
+      CFASTBuilder cfastBuilder,
+      Provider<CobolLanguageClient> clientProvider) {
```

Comment:
I think it would be better to just add a new constructor and make the CFAST dependency optional.
As far as I can tell, this is the source of the need to bind the CFACTBuilder in daco and idms code.

Reply:
I will think how to do it